src/shell/dotnet/Shell/Fdc3/ResolverUi/Fdc3ResolverUi.xaml.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
+using System.Windows;
+using System.Windows.Input;
+using Finos.Fdc3;
+
+namespace MorganStanley.ComposeUI.Shell.Fdc3.ResolverUI;
+
+/// <summary>
+/// Interaction logic for Fdc3ResolverUi.xaml
+/// </summary>
+public partial class Fdc3ResolverUI : Window, IDisposable
+{
+    private readonly Fdc3ResolverUIViewModel _viewModel;
+
+    public IAppMetadata? AppMetadata { get; internal set; }
+    public CancellationToken UserCancellationToken => _viewModel.UserCancellationToken;
+
+    public Fdc3ResolverUI(IEnumerable<IAppMetadata> apps)
+    {
+        InitializeComponent();
+
+        _viewModel = new Fdc3ResolverUIViewModel(apps);
+        DataContext = _viewModel;
+        _viewModel.SizeChanged += OnSizeChanged;
+    }
+
+    private void OnSizeChanged(object? sender, PageSizeChangedEventArgs e)
```

Comment:
Instead of using events have you tried simply binding Width/Height to a property?